pipeline/psth.py

```diff
@@ -367,71 +367,55 @@ def make(self, key):
 
         # retrieving the spikes of interest,
         spikes_q = ((ephys.Unit.TrialSpikes & key)
-                    & (experiment.BehaviorTrial()
+                    * (experiment.BehaviorTrial()
                        & {'task': 'audio delay'}
                        & {'early_lick': 'no early'}
                        & {'outcome': 'hit'}) - experiment.PhotostimEvent)
 
-        # and their corresponding behavior,
-        lr = ['left', 'right']
-        behav = (experiment.BehaviorTrial & spikes_q.proj()).fetch(
-            order_by='trial asc')
-        behav_lr = {k: np.where(behav['trial_instruction'] == k)[0] for k in lr}
-
-        if egpos['hemisphere'] == 'left':
-            behav_i = behav_lr['left']
-            behav_c = behav_lr['right']
-        else:
-            behav_i = behav_lr['right']
-            behav_c = behav_lr['left']
-
-        # constructing a square, nan-padded trial x spike array
-        spikes = spikes_q.fetch(order_by='trial asc')
-        ydim = max(len(i['spike_times']) for i in spikes)
-        square = np.array(
-            np.array([np.concatenate([st, pad])[:ydim]
-                      for st, pad in zip(spikes['spike_times'],
-                                         repeat([math.nan]*ydim))]))
-
-        # with which to calculate the selectivity over the given period
-        period = (experiment.Period & key).fetch1()
-
-        # by determining the period boundaries,
-        bounds = (period['period_start'], period['period_end'])
-
-        # masking the appropriate spikes,
-        lower_mask = np.ma.masked_greater_equal(square, bounds[0])
-        upper_mask = np.ma.masked_less_equal(square, bounds[1])
-        inrng_mask = np.logical_and(lower_mask.mask, upper_mask.mask)
+        if not spikes_q:  # no spikes found
+            self.insert1({**key, 'period_selectivity': 'non-selective'})
+            return
 
-        # computing their spike rate,
-        rsum = np.sum(inrng_mask, axis=1)
-        dur = bounds[1] - bounds[0]
-        freq = rsum / dur
+        # retrieving event times
```

Comment:
wondering if this would have worked with multidimentional 'bounds', but in any event.